app/assets/javascripts/sitewide/file_preview.js

```diff
@@ -448,6 +449,7 @@ var FilePreviewModal = (function() {
             if (!readOnly && data.editable) {
               modal.find('.file-edit-link').css('display', '');
               modal.find('.file-edit-link').off().click(function(ev) {
+                $.post('/files/' + data.id + '/start_edit');
```

Comment:
Shouldn't we add this activity on `image_loaded` event or even better when user started drawing/editing the image to make sure user doesn't just mistakenly click on the edit button?

Reply:
I think it will be very complicated. Now if user don't make any changes, than we will not have activity that he edited image. Cause second activity triggered only if he applied changes.

Reply:
Yeah I meant to put into 285th line:
```
    imageEditor.on('image_loaded', () => {
      $('.file-save-link').css('display', '');
    });

```
but if you think this does not bring anything to the table, then discard this comment.

Reply:
It is very close actions, for average images less than second.